backend/main/models/proposal.go

```diff
@@ -291,3 +298,85 @@ func GetActiveStrategiesForCommunity(db *s.Database, communityId int) ([]string,
 
 	return strategies, nil
 }
+
+func handleCancelledProposal(db *s.Database, proposalId int) error {
+
+	// Delete All votes for cancelled proposal
+	_, err := db.Conn.Exec(db.Context, `
+		DELETE FROM votes WHERE proposal_id = $1
+	`, proposalId)
+
+	if err != nil {
+		return err
+	}
+
+	// Delete All non-streak achievements
+	_, err = db.Conn.Exec(db.Context, `
+		DELETE FROM user_achievements WHERE $1 = ANY (proposals) AND achievement_type != 'streak'
+	`, proposalId)
+
+	if err != nil {
+		return err
+	}
+
+	var achievements = []struct {
+		Id int64 `json:"id"`
+		Proposals []int64 `json:"proposals"`
+		Details string `json:"details"`
+	}{}
+
+	// handle streaks
+	err = pgxscan.Select(db.Context, db.Conn, &achievements, `
+		SELECT id, proposals, details FROM user_achievements WHERE $1 = ANY (proposals) AND achievement_type = 'streak'
+	`, proposalId)
+
+	if err != nil && err.Error() != pgx.ErrNoRows.Error() {
+		return err
+	}
```

Comment:
I think we do need it, because if there are no rows, then achievements will be an empty slice, the for loop after won't execute and it will return nil. If we remove that then we would be handling NoRows as an error, which we don't want because it just means that there was nothing to update.